test/functional/provider/python3_spec.lua

```diff
@@ -13,7 +13,7 @@ local dedent = helpers.dedent
 
 do
   clear()
-  local reason = missing_provider('python3')
+  local reason = missing_provider('python')
```

Comment:
`python3` or `python` (consistency)? Or is it like getting prepared for python4 in the future?

Reply:
Shall I revert this?

Reply:
In general I think it would be good to rename the name of provider `python3` to `python` in this chance, while keeping the backward compatibility (e.g. `g:python3_host_prog`, etc.)

Reply:
If you say this is fine then it's fine. If anyone disagrees they can bring it up themselves.